audio-worklet/src/noisegate-script-processor.js

```diff
@@ -95,18 +107,12 @@ class NoiseGate {
     }
 
     let envelope = this.detectLevel_(this.channel_);
-
-    // For sine waves, the envelope eventually reaches an average power of
-    // a^2 / 2. Sine waves are therefore scaled back to the original amplitude,
-    // but other waveforms or constant sources can only be approximated.
-    for (let k = 0; k < this.envelope_.length; k++) {
-      envelope[k] =  this.toDecibel_(Math.sqrt(envelope[k]) * Math.sqrt(2));
-    }
+    envelope = this.convertEnvelopeToSineNormalizedDecibelScale_(envelope);
 
     for (let i = 0; i < inputBuffer.numberOfChannels; i++) {
       let input = inputBuffer.getChannelData(i);
       let output = event.outputBuffer.getChannelData(i);
-      let weights = this.computeGain_(envelope);
+      let weights = this.computeWeights_(envelope);
```

Comment:
Is there any need to compute weights again for each channel?  The weights only depend on the envelope, right?

If this is right, then computeWeights_ can do the scaling by sqrt(2) itself instead of having to do it in line 110 above.